Sources/Sentry/include/SentryTracer.h

```diff
@@ -70,55 +64,17 @@ static NSTimeInterval const SentryTracerDefaultTimeout = 3.0;
                                        hub:(nullable SentryHub *)hub;
 
 /**
- * Init a SentryTracer with given transaction context, hub and whether the tracer should wait
- * for all children to finish before it finishes.
- *
- * @param transactionContext Transaction context
- * @param hub A hub to bind this transaction
- * @param waitForChildren Whether this tracer should wait all children to finish.
- *
- * @return SentryTracer
- */
-- (instancetype)initWithTransactionContext:(SentryTransactionContext *)transactionContext
-                                       hub:(nullable SentryHub *)hub
-                           waitForChildren:(BOOL)waitForChildren;
-
-/**
- * Init a SentryTracer with given transaction context, hub and whether the tracer should wait
- * for all children to finish before it finishes.
- *
- * @param transactionContext Transaction context
- * @param hub A hub to bind this transaction
- * @param profilesSamplerDecision Whether to sample a profile corresponding to this transaction
- * @param waitForChildren Whether this tracer should wait all children to finish.
- * @param timerWrapper A writer around NSTimer, to make it testable
- *
- * @return SentryTracer
- */
-- (instancetype)initWithTransactionContext:(SentryTransactionContext *)transactionContext
-                                       hub:(nullable SentryHub *)hub
-                   profilesSamplerDecision:
-                       (nullable SentryProfilesSamplerDecision *)profilesSamplerDecision
-                           waitForChildren:(BOOL)waitForChildren
-                              timerWrapper:(nullable SentryNSTimerWrapper *)timerWrapper;
-
-/**
- * Init a SentryTracer with given transaction context, hub and whether the tracer should wait
- * for all children to finish before it finishes.
+ * Init a SentryTracer with given transaction context and hub and set other fields by default
  *
  * @param transactionContext Transaction context
  * @param hub A hub to bind this transaction
- * @param profilesSamplerDecision Whether to sample a profile corresponding to this transaction
- * @param idleTimeout The idle time to wait until to finish the transaction.
+ * @param configuration Configuration on how SentryTracer will behave
  *
  * @return SentryTracer
  */
 - (instancetype)initWithTransactionContext:(SentryTransactionContext *)transactionContext
                                        hub:(nullable SentryHub *)hub
-                   profilesSamplerDecision:
-                       (nullable SentryProfilesSamplerDecision *)profilesSamplerDecision
-                               idleTimeout:(NSTimeInterval)idleTimeout
-                      dispatchQueueWrapper:(SentryDispatchQueueWrapper *)dispatchQueueWrapper;
+                             configuration:(nullable SentryTracerConfiguration *)configuration;
```

Comment:
I like this refactor, but my one suggestion is to make this nonnull and provide a `SentryTracerConfiguration.default` that can be supplied. That makes it more obvious at callsites of this constructor what is going on, rather than having to go to the implementation of this init to see what happens if `nil` is passed in. This is also a common pattern throughout Cocoa APIs.